compiler/src/main/scala/org/scalajs/core/compiler/JSEncoding.scala

```diff
@@ -61,7 +61,7 @@ trait JSEncoding extends SubComponent { self: GenJSCode =>
       longName = base+"$"+suffix
     }
     usedLocalNames += longName
-    longName
+    mangleJSName(longName)
```

Comment:
No, it is not a bug, because all other uses of `mangleJSName` are accompanied by some suffix. For example for methods you have a suffix `__xyz`. For fields you have a suffix `$f` or `$1`. Moreover, all of these uses are for identifiers in the *property* namespace of JS (for after a `.`), where reserved identifiers can be legally used anyway (but not keywords). This is why only `freshLocalName` handles reserved identifiers specially.